components/TextFields/src/MDCTextInputController.h

```diff
@@ -0,0 +1,200 @@
+
+ Default is black with Material Design hint text opacity.
+ */
+@property(nonatomic, nullable, strong) UIColor *inlinePlaceholderColor NS_SWIFT_NAME(inlineColor)
+    UI_APPEARANCE_SELECTOR;
+
+/*
+ Indicates whether the alert contents should automatically update their font when the device’s
+ UIContentSizeCategory changes.
+
+ This property is modeled after the adjustsFontForContentSizeCategory property in the
+ UIConnectSizeCategoryAdjusting protocol added by Apple in iOS 10.0.
+
+ Default value is NO.
+ */
+@property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)
+    BOOL mdc_adjustsFontForContentSizeCategory UI_APPEARANCE_SELECTOR;
+
+/** The behavioral style applied to the text input. */
+@property(nonatomic, assign)
+    MDCTextInputPresentationStyle presentationStyle NS_SWIFT_NAME(presentation);
+
+/** The text input the controller is affecting. */
+@property(nonatomic, nullable, strong) UIView<MDCTextInput> *textInput NS_SWIFT_NAME(input);
+
+/**
+ Controls when the underline will be shown.
+
+ The underline is an overlay that can be hidden depending on the editing state of the input text.
+
+ Default is UITextFieldViewModeAlways.
+ */
+@property(nonatomic, assign) UITextFieldViewMode underlineViewMode NS_SWIFT_NAME(underlineMode)
+    UI_APPEARANCE_SELECTOR;
+
+/**
+ Convenience init. Never fails.
+
+ @param input An MDCTextInput this controller will manage.
+ */
+- (nonnull instancetype)initWithTextInput:(nullable UIView<MDCTextInput> *)input
+    NS_SWIFT_NAME(init(input:));
+
+/**
+ Sets the state of the controller by setting the values of properties errorText and
+ errorAccessibilityValue.
+
+ The value of errorText controls the state of the text input.
+
+ @param errorText               The error text to be shown as underline text. (Copied.)
+ @param errorAccessibilityValue Optional override of default underline text accessibility value.
+                                (Copied.)
+
+ When errorText != nil, the text input is in an error state:
+ - The error text appears in the underline text with the errorColor as text color.
+ - The input rectangle's underline, placeholder and character are colored to the errorColor.
+
+ When errorText == nil, the text input is not in error state:
+ - The underline text is restored to the color and value it was.
+ - The underline color and height is restored.
+ - The placeholder text is restored to the color it was.
+ - The character count text is restored to the color it was.
+
+ Setting errorText to an empty string (@"") will put the input in error state but the underline text
+ will be empty. It will color (if visible) the underline, the placeholder, and the character
+ count in the errorColor.
+
+ Setting errorAccessibilityValue when errorText == nil has no effect.
+
+ Note: The characterCountMax property also affects these same MDCTextInput properties.
+ */
+- (void)setErrorText:(nullable NSString *)errorText
```

Comment:
The delegate is the regular UITextFieldDelegate. I've added a mention to it in the documentation.